etcd/cmd/microshift-etcd/run.go

```diff
@@ -97,17 +112,66 @@ func (s *EtcdService) Run() error {
 	}
 	<-e.Server.ReadyNotify()
 
+	// Start up the defrag controller.
+	defragCtx, defragShutdown := context.WithCancel(context.Background())
+	go s.defragController(defragCtx, e.Server.Backend())
+
 	// Wait to be stopped.
 	sigTerm := make(chan os.Signal, 1)
 	signal.Notify(sigTerm, os.Interrupt, syscall.SIGTERM)
 	sig := <-sigTerm
 	klog.Infof("microshift-etcd received signal %v - stopping", sig)
 
+	// Shutdown the defrag controller.
+	defragShutdown()
+
 	e.Server.Stop()
 	<-e.Server.StopNotify()
 	return nil
 }
 
+func (s *EtcdService) defragController(ctx context.Context, be backend.Backend) error {
+	// If we were told to, go ahead and do a defragment now.
+	if s.doStartupDefrag {
+		if err := be.Defrag(); err != nil {
+			err = fmt.Errorf("initial defragmentation failed: %v", err)
+			klog.Error(err)
+			return err
```

Comment:
This error is not caught because 
```
go s.defragController(defragCtx, e.Server.Backend())
```
Maybe startup defrag should be outside defragController?